Makefile

```diff
@@ -207,8 +207,14 @@ test-pyright : $(VENV) $(DEFAULT_TARGET)
 test-pytest : $(VENV) $(DEFAULT_TARGET)
 	$(ACTIVATE_VENV)
 
-	pytest $(PYTEST_FLAGS)
-	coverage html --data-file=$(REPORTS_DIR)/pytest/.coverage
+	pytest $(PYTEST_FLAGS) \
+		&& PYTEST_SUCCESS=0 \
+		|| PYTEST_SUCCESS=$$?
+
+	-coverage html --data-file=$(REPORTS_DIR)/pytest/.coverage
+	-junit2html $(REPORTS_DIR)/pytest/pytest.xml $(REPORTS_DIR)/pytest/pytest.html
+
+	exit $$PYTEST_SUCCESS
```

Comment:
The nitiest of picks, `PYTEST_SUCCESS` suggests a boolean but you're storing and returning the full exit code...